pkg/manifests/manifests_test.go

```diff
@@ -1469,6 +1474,31 @@ func TestPrometheusQueryLogFileConfig(t *testing.T) {
 	}
 }
 
+func TestPrometheusK8sConfigurationBodySizeLimitDefault(t *testing.T) {
+	c, err := NewConfigFromString(`
+prometheusK8s:
+    enforceBodySizeLimit: "10MB"
+  `)
+
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	f := NewFactory("openshift-monitoring", "openshift-user-workload-monitoring", c, defaultInfrastructureReader(), &fakeProxyReader{}, NewAssets(assetsPath), &APIServerConfig{}, nil)
+	p, err := f.PrometheusK8s(
+		&v1.Secret{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+		&v1.ConfigMap{ObjectMeta: metav1.ObjectMeta{Name: "foo"}},
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// the body size limit value is not set at configuration parsing time.
+	if p.Spec.EnforcedBodySizeLimit != "" {
+		t.Fatalf("EnforcedBodySizeLimit is not configured correctly, expected empty string, but got %v", p.Spec.EnforcedBodySizeLimit)
+	}
```

Comment:
it should be the opposite

Reply:
will add LoadEnforcedBodySizeLimit after config parsing and test the automatic deduced bodysize limit.